src/components/CrucibleFilter.vue

```diff
@@ -1,78 +1,98 @@
+        </span>
+      </div>
+
+      <!-- Selecting Filter tags from the Collection list -->
+      <div class="crucible-filters">
+        <div
+          v-for="(items, key) in taxonomyGroups"
+          :key="key"
+          class="crucible-filter-dropdown"
+        >
+          <h4 @click="toggleDropdown(key)">
+            <span> {{ key }}</span>
+            <CollapseBtn :show-dropdown="showDropdown[key]" />
+          </h4>
+          <ul v-show="showDropdown[key]" class="crucible-filter-dropdown-menu">
+            <li
+              v-for="(item, index) in items"
+              :key="index"
+              :class="
+                itemNames.includes(Object.keys(item)[0])
+                  ? 'selected-filter-tag'
+                  : ''
+              "
+              @click="getFilterTag(key, Object.keys(item)[0])"
+            >
+              {{ Object.keys(item)[0] }}
+              <span>({{ Object.values(item)[0] }})</span>
+            </li>
+          </ul>
+        </div>
       </div>
     </div>
-    <button class="filter-btn" @click="applyFilter">Apply</button>
-    <button class="filter-btn" @click="resetFilter">Empty</button>
+
+    <!-- Controlling Filter panel visibility -->
+    <button
+      :class="
+        showFilter
+          ? 'crucible-filter-control svg-background'
+          : 'crucible-filter-control'
+      "
+      @click="showFilter = !showFilter"
+    >
+      <span v-if="!showFilter">Filters</span>
+    </button>
   </div>
 </template>
 
 <script setup lang="ts">
-import { ref } from "vue";
+import { ref, computed } from "vue";
+import CollapseBtn from "./CollapseBtn.vue";
+import FilterButton from "./FilterButton.vue";
 //ToDo: inject the taxonomyTags from the Crucible Main platform
-import { taxonomyTags } from "@/resources";
+import { taxonomyGroups } from "./DataAccessLayer";
 
+const showFilter = ref<boolean>(false);
 const showDropdown = ref({} as Record<string, boolean>);
 const filterTagArray = ref([] as string[]);
 
-// Convert to Taxonomy Category Object array
-const category = taxonomyTags.reduce(
-  (acc, tag) => {
-    const [key, value] = tag.split(":");
-    if (!acc[key]) {
-      acc[key] = [];
-    }
-    acc[key].push(value.replace("_", " "));
-    return acc;
-  },
-  {} as Record<string, string[]>,
-);
+// Convert to Taxonomy Object array with {taxonomy:{tag:resourceSize}}
 
 // function to toggle drop down
 const toggleDropdown = (key: string) => {
   showDropdown.value[key] = !showDropdown.value[key];
-  Object.keys(showDropdown.value).forEach((k) => {
-    if (k !== key) {
-      showDropdown.value[k] = false;
-    }
-  });
 };
 
 // Filter tags send to back end to filter resource
 const getFilterTag = (key: string, tag: string) => {
-  showDropdown.value[key] = !showDropdown.value[key];
   if (!filterTagArray.value.includes(`${key}:${tag.replace(" ", "_")}`)) {
     filterTagArray.value.push(`${key}:${tag.replace(" ", "_")}`);
   }
```

Comment:
I think in terms of readability, assigning a variable here would be a bit better. i.e.
```ts
  const formattedTag = `${key}:${tag.replace(" ", "_")}`;
  if (!filterTagArray.value.includes(formattedTag)) {
    filterTagArray.value.push(formattedTag);
  }
```

Reply:
fixed